### DISCUSSION TITLE
Make error messages more explicit

### DISCUSSION THREAD

**Discussion by claudiux** [linuxmint/discussions#48] [Development]
In the `~/.xsession-errors` file you'll find messages you don't know what to do with, because you don't know where the error originated. For example, the following:
```
(cinnamon:2066): Gjs-WARNING **: 09:42:39.463: Some code accessed the property 'Util' on the module 'main'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
```

or

```
(cinnamon:2066): Gjs-WARNING **: 09:23:51.996: Some code called array.toString() on a Uint8Array instance. Previously this would have interpreted the bytes of the array as a string, but that is nonstandard. In the future this will return the bytes as comma-separated digits. For the time being, the old behavior has been preserved, but please fix your code anyway to explicitly call ByteArray.toString(array).
(Note that array.toString() may have been called implicitly.)
0 loadFileContent/<() ["/usr/share/cinnamon/js/misc/fileUtils.js line 211 > Function":142:33]
```

Is it possible to make these messages more explicit and useful for developers, please?

**Comment by mtwebster** [comment]
I can have a look at the first.

The second is very frustrating, I agree.. the "142:33" I corresponds to the line:character in the faulty file, but you have to know what file it is, which isn't always apparent. I'm pretty sure we lost a lot of useful traces due to this small commit: https://github.com/linuxmint/cinnamon/commit/f01bd803d5ae8978d934e1a84d07222e6ec866c4).

We've looked at trying to rework that, but so far everyone who's tried wanted to bash their head in after looking into it. I spent some time today trying to see what I could do for it but no luck.

**Comment by claudiux** [comment]
Thank you for searching!

**Comment by claudiux** [comment]
@mtwebster 
I've found the source of the second type of error message: the `notes@schorschii` desklet. I'm going to make a PR.

As for the first type of error message, I've modified the `fileUtils.js` script, replacing some 'const' and 'let' by 'var' and I don't see any more error messages! I'm also going to make a PR.

**Reply by rcalixte** [reply]
I've opened PRs for ES6 compatibility in the past as I've encountered exceptions in the logs myself as well.

@mtwebster @clefebvre Would it be worth a mini-initiative after the upcoming releases are completed to get that code cleanup done? Or are adhoc PRs just fine for now?

**Comment by claudiux** [comment]
https://github.com/linuxmint/cinnamon-spices-desklets/pull/855 and https://github.com/linuxmint/cinnamon/pull/11715

**Comment by claudiux** [comment]
@mtwebster 

> I'm pretty sure we lost a lot of useful traces due to this small commit: https://github.com/linuxmint/cinnamon/commit/f01bd803d5ae8978d934e1a84d07222e6ec866c4).

I tried to get this "small" commit under control, but I didn't see anything that made the name of the offending file disappear from the error messages.
